File: benchmarks/cli/pipe_10k.nim
import std/[times, strutils, sequtils]
var lines: seq[string]
for i in 0..<10000:
  lines.add("line " & $i & ": the quick brown fox")
let start = cpuTime()
let result = lines.mapIt(it.toUpperAscii())
echo int((cpuTime() - start) * 1_000_000)

File: benchmarks/cli/pipe_10k.rb
lines = Array.new(10000) { |i| "line #{i}: the quick brown fox" }
start = Process.clock_gettime(Process::CLOCK_MONOTONIC, :microsecond)
result = lines.map(&:upcase)
finish = Process.clock_gettime(Process::CLOCK_MONOTONIC, :microsecond)
puts(finish - start)
